pagopa/io-app
# chore: [LLK-66] LP key login flow

## Short description
Added control flows to login to support lollipop.

## List of changes proposed in this pull request
- [locales/en/index.yml, locales/it/index.yml]: added sentences for unsupported devices page.
- package.json: updated backend api.
- ts/features/lollipop/__tests__/login.test.ts: added tests for every any new login flow.
- ts/features/lollipop/saga/index.ts: 
      - added saga to check data consistency between server and device.
      - added function that shows a new screen for unsupported devices.
- ts/features/lollipop/screens/FailureScreen.tsx: new screen for unsupported devices.
- [ts/navigation/AppStackNavigator.tsx, ts/navigation/params/AppParamsList.ts, ts/navigation/routes.ts]: added route for the new screen.
- ts/sagas/__tests__/initializeApplicationSaga.test.ts: update.
- ts/sagas/commons/index.ts: added function that properly restart the app.
- ts/sagas/startup.ts: 
       - added check with server for data consistency.
       - added check to display the new screen for unsupported devices.
- ts/sagas/startup/watchSessionExpiredSaga.ts: refactored to use the new function to restart the application.
- ts/screens/messages/MessagesHomeScreen.tsx: edited the key that contains the unsupported device message, to be clearer with all new keys.
- ts/store/reducers/authentication.ts: edited TokenName to not consider `lollipopAssertionRef` from payload.

## How to test

Make sure you are using a dev server version that includes this branch:
[LLK-66](https://github.com/pagopa/io-dev-api-server/tree/LLK-66-lp-auth)

- Run test files.

### Supported device
- Log in -> reset dev-server(so it loses "stored" info) -> terminate and reopen the app: it should disconnect you due to data inconsistency.
- run dev-server -> run the app with a clean installation: it should work as usual.

### Unsupported device
- Log in with an old app version -> reopen the app with this branch: it should log you out due to the absence of the public key and show you the unsupported device screen.
- Open the app with a clean installation: it should show you the unsupported device screen.



[LLK-66]: https://pagopa.atlassian.net/browse/LLK-66?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
